sdk/dotnet/Pulumi/Resources/DictionaryResource.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2016-2020, Pulumi Corporation
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace Pulumi
+{
+    public class DictionaryResource : Resource
+    {
+#pragma warning disable RS0022 // Constructor make noninheritable base class inheritable
+        public DictionaryResource(string type, string name, ImmutableDictionary<string, object?> args, ResourceOptions? options = null)
+            : base(type, name, true, new DictionaryResourceArgs(args), options ?? new ResourceOptions())
+#pragma warning restore RS0022 // Constructor make noninheritable base class inheritable
+        {
+        }
```

Comment:
so.... i started out likeing this, but now i don't like it (and we shoudl discuss this). This distinctly moves the .net sdk away from the rest of our SDKs.  It's really a strange thing tbh as this would no end up in things like checkpoints, and it contains things that are pretty strange (like URNs).  

Sorry to torpedo this at the last minute... but we should talk to find another way here.